### ISSUE TITLE
Submission attachment is shown as downloadable even when it's missing

### ISSUE THREAD

**Issue by lognaturel** [getodk/central#706]
#### Problem description
Submission attachment is shown as downloadable even when it's missing

#### URL of the page
https://test.getodk.cloud/#/projects/149/forms/annotate_default/submissions

#### Steps to reproduce the problem
Make a submission with a missing attachment (for example, in Enketo, annotate an image, scale down the canvas, submit), view the submission table

#### Screenshot

<img width="928" alt="Screenshot 2024-09-10 at 8 42 56 AM" src="https://github.com/user-attachments/assets/870b89b5-3750-4132-a10b-29bb3e29330d">

#### Expected behavior
I would expect the missing file to be displayed differently. If it's not practical to change, we could just remove the "File was submitted." from the tooltip.

#### Central version shown in version.txt
```
versions:
f21b33acc8dae215e54485849e7dd59167e6f433 (v2024.1.0)
 3fb0c22b1cbdc3a6004963afcc3847a82c09307d client (v2024.1.0)
 b4754cf52bfa64b1ca841bc9ccb64a38726398e8 server (v2024.1.0)
```

#### Browser version
Firefox

#### Around when did you see the problem (in UTC)?

3:44 PM
Tuesday, September 10, 2024

**Comment by matthew-white** [comment]
> I would expect the missing file to be displayed differently. If it's not practical to change, we could just remove the "File was submitted." from the tooltip.

I agree that it would be ideal for the missing file to be displayed differently. Currently, that's not information available to Frontend via the OData feed. Backend returns `attachmentsExpected` and `attachmentsPresent`, so Frontend can determine the number of missing files, but not which files are missing. I'm not sure how best for Backend to return that information, though I reckon there's got to be some way. But certainly it'd be very easy to remove that text from the tooltip.

**Comment by matthew-white** [comment]
Just to throw out another option, it wouldn't be hard to have Frontend preflight a HEAD request for the submission attachment. If it's a 404, Frontend could replace the checkmark with an X (or something) along with an explanatory tooltip.

**Comment by sadiqkhoja** [comment]
Another option: Update odata to include `attachments` metadata (name and exists) as [collection](https://docs.oasis-open.org/odata/odata/v4.0/os/part3-csdl/odata-v4.0-os-part3-csdl.html#_Toc372794040) type under `__system` property

One more option: introduce attachments as NavigationProperty similar to "repeats table" and enhance `$expand` to be able expand attachments 

By updating Odata, we would be able show correct icon on the load, without making requests for each Submission

**Comment by matthew-white** [comment]
That does sound like a more standard approach. Attachments are a subresource of submissions, so I think it makes sense to include more detailed information about them in OData.

How hard do you think that change would be in Backend? Part of why I liked the idea of the HEAD request was that it wouldn't require a change to Backend. But it is an additional request, and things would be simpler in Frontend if Frontend could just tell from OData whether a particular attachment exists.
